scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/StaxJobFactory.java

```diff
@@ -537,6 +560,55 @@ public JobType getType() {
 
     }
 
+    private static synchronized Map<String, JobVariable> getConfiguredGlobalJobVariables() {
+        if (globalVariables == null) {
+            globalVariables = new LinkedHashMap<>();
+            String path = PASchedulerProperties.getAbsolutePath(PASchedulerProperties.GLOBAL_VARIABLES_CONFIGURATION.getValueAsString());
+            logger.info("Loading global variables from file: " + path);
+            Properties props = new Properties();
+            try (InputStream fis = new FileInputStream(path)) {
+                props.load(fis);
+            } catch (IOException e) {
+                logger.error("Global Variable Configuration file: " + path + " cannot be read.", e);
+                return globalVariables;
+            }
+            Map<String, String> propertyMap = (Map) props;
+            for (Map.Entry<String, String> var : propertyMap.entrySet()) {
+                globalVariables.put(var.getKey(), new JobVariable(var.getKey(), var.getValue()));
+            }
+        }
+        return new LinkedHashMap<>(globalVariables);
```

Comment:
Do we need to set and return globalVariables?

Reply:
it is important to return a clone, otherwise, changes to the map are persisted (static variable) on subsequent workflow parsing.

getConfiguredGlobalJobVariablesAsReplacementMap() uses directly the static variable, and return as well a new map

Reply:
I agree that in the second method, I could check that the variable is initialized instead of calling getConfiguredGlobalJobVariables() systematically